### ISSUE TITLE
augmentation: RandomResizedCrop slice mode resamples masks with align_corners=False when the mask override passes align_corners=None

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4854]
## Summary

`GeometricAugmentationBase2D.apply_transform_mask` replaces an `align_corners=None` mask override with the module's `align_corners`, except in `cropping_mode="slice"`, where it keeps `None`. That exception existed because slice mode passed `align_corners` to `F.interpolate`, which rejects it for nearest (#4802). #4852 makes `RandomResizedCrop` drop `align_corners` for nearest itself, so the exception no longer prevents a crash. It now only affects the interpolating modes. With a bilinear or bicubic mask override and `align_corners=None`, slice mode resizes the mask as `align_corners=False` (interpolate's reading of `None`), while the image uses the module's value, which defaults to `True`. Mask and image then stop lining up. Resample mode does not have this problem.

## Reproduction (torch 2.14.0, CPU, kornia main at 9c39a1096)

```python
import torch, kornia.augmentation as K
from kornia.constants import DataKey, Resample

torch.manual_seed(0)
img = torch.rand(2, 1, 16, 16)
for mode in ["slice", "resample"]:
    seq = K.AugmentationSequential(
        K.RandomResizedCrop((11, 11), p=1.0, cropping_mode=mode, align_corners=True),
        data_keys=["input", "mask"],
        extra_args={DataKey.MASK: {"resample": Resample.BILINEAR, "align_corners": None}},
    )
    out, mask = seq(img, img.clone())
    print(mode, torch.equal(out, mask), (out - mask).abs().max().item())
# slice False 0.107...
# resample True 0.0
```

## Expected

`align_corners=None` in the mask override should mean "the module's value" in both cropping modes, as `apply_transform_mask`'s own comment says. After #4852, the slice-mode `pass` branches in `apply_transform_mask` and `inverse_masks` can be removed, and nearest still works. With that change, the repro prints `slice True`, and the `tests/augmentation/container` and `test_conventions_geometric_crop.py` suites pass unchanged (874 passed). No existing test pins the current behaviour, so the fix needs one. It also changes outputs for slice-mode bilinear/bicubic mask overrides with `None`, so the fragment should say so.

Found while reviewing #4852.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).

**Comment by adimalkar** [comment]
Reproduced on `main` (`7ddf731b`, after #4852), CPU float32, with the snippet above:

```
slice False 0.10706233978271484
resample True 0.0
```

I'm working on the fix you describe — dropping the slice-mode `pass` branches in `apply_transform_mask` and `inverse_masks` — plus a test that pins the slice-mode mask against the image for the interpolating modes, and a fragment noting the output change. PR shortly.

(I use an AI assistant for this; the reproduction and test results are from my local runs.)
